lib/fs/casefs.go

```diff
@@ -445,8 +450,8 @@ func (r *defaultRealCaser) dropCache() {
 // getExpireAdd gets an entry for the given key. If no entry exists, or it is
 // expired a new one is created and added to the cache.
 func (r *defaultRealCaser) getExpireAdd(key string) *caseNode {
-	r.mut.Lock()
-	defer r.mut.Unlock()
+	r.cache.mut.Lock()
+	defer r.cache.mut.Unlock()
```

Comment:
optional, nit: Now that the lock is part of the cache again, might be more readable for this to be a method on `caseCache`  (revert to state before the referenced change) passing the filesystem as a parameter.